grabbit/core.py

```diff
@@ -1,26 +1,164 @@
+                                                optional_chunk)
+                else:
+                    new_path = new_path.replace('[%s]' % optional_pattern,
+                                                '')
+
+            new_path = self.replace_entities(new_path)
+            if new_path:
+                return new_path
+
+    def write_file(self, path_patterns=None, symbolic_link=False,
+                   root=None, conflicts='fail', copy_into_dir=True):
+        """
+        Uses provided filename patterns to write this file to a new path, given
+        this file's corresponding entity values.
+
+        Args:
+            path_patterns (str, list): One or more filename patterns to write
+                the file to.
+            symbolic_link (bool): Whether to copy the file as a symbolic link
+                or a deep copy.
+            root (str): Optional root directory that all patterns are relative
+                to. Defaults to current working directory.
+            conflicts (str): One of 'fail', 'skip', 'overwrite', or 'append'
+                that defines the desired action when the output path already
+                exists. 'fail' raises an exception; 'skip' does nothing;
+                'overwrite' overwrites the existing file; 'append' adds a suffix
+                to each file copy, starting with 0. Default is 'fail'.
+            copy_into_dir (bool): If a path pattern is a directory, this flag
+                indicates whether to use the original basename and copy into
+                the directory. Otherwise, will default to the behavior
+                specified by the `conflicts` parameter.
+        """
+
+        new_filename = self.get_path(path_patterns=path_patterns)
+        if not new_filename:
+            return
+
+        if not root and not isabs(new_filename):
+            root = os.getcwd()
+
+        new_filename = join(root, new_filename)
+
+        if new_filename == self.path:
+            return
+
+        if not exists(dirname(new_filename)):
+            os.makedirs(dirname(new_filename))
+
+        if new_filename[-1] == os.sep and copy_into_dir:
+            new_filename += self.filename
+
+        if exists(new_filename) or islink(new_filename):
+            if conflicts == 'fail':
+                msg = 'A file at path {} already exists.'
+                raise ValueError(msg.format(new_filename))
+            elif conflicts == 'skip':
+                msg = 'A file at path {} already exists, skipping writing file.'
+                logging.warn(msg.format(new_filename))
+                return
+            elif conflicts == 'overwrite':
+                if isdir(new_filename):
+                    logging.warn('New path is a directory, not going to '
+                                 'overwrite it, skipping instead.')
+                    return
+                os.remove(new_filename)
+            elif conflicts == 'append':
+                i = 1
```

Comment:
Looks like it starts at 1, but docstring says 0. Either is fine, just make sure it's consistent.